File: frontend/src/type/booking.ts
export interface BookingLocationState {
    roomId: string | number;
    checkIn: string;  // Định dạng từ widget: "2026-11-15T14:00"
    checkOut: string; // Định dạng từ widget: "2026-11-17T12:00"
    price: number;
}

export interface PaymentFormData {
    customerName: string;
    phone: string;
    email: string;
    nationality: string;
    isBookingForOthers: boolean;
    checkInTimeWindow: string;
    note: string;
    paymentMethod: 'Thẻ Quốc Tế' | 'Ví Điện Tử' | 'Chuyển khoản' | 'Thanh toán tại Khách sạn';
    cardNumber?: string;
    cardExpiry?: string;
    cardCvv?: string;
    promoCode: string;
    agreedToTerms: boolean;
}

// Thêm vào file type/booking.ts hoặc nơi bạn quản lý Type
export interface RoomItemRegister {
    roomTypeId: number;
    checkIn: string;
    checkOut: string;
    quantity: number;
}

export interface BookingRequestPayload {
    customerName: string;
    phone: string;
    email: string;
    nationality: string;
    specialRequests: string;
    paymentMethod: string;
    promoCode: string;
    rooms: RoomItemRegister[];
}

File: backend/src/main/java/project/backend/entities/Booking.java
package project.backend.entities;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;
import project.backend.eNum.BookingStatus;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "bookings")
@Data
public class Booking {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(unique = true, nullable = false)
    private String bookingCode;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user; // Khách đã có tài khoản

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_type_id", nullable = false)
    private RoomType roomType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "room_id")
    private Room room; // Phòng vật lý cụ thể (Staff gắn vào lúc check-in)

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "promotion_id")
    private Promotion promotion;

    @Column(nullable = false)
    private LocalDate checkIn;

    @Column(nullable = false)
    private LocalDate checkOut;

    @Column(nullable = false)
    private Integer guestsCount;

    @Column(nullable = false)
    private BigDecimal basePrice;

    private BigDecimal discountAmount = BigDecimal.ZERO;

    @Column(nullable = false)
    private BigDecimal totalPrice;

    @Enumerated(EnumType.STRING)
    @JdbcTypeCode(SqlTypes.NAMED_ENUM)
    private BookingStatus status = BookingStatus.pending;

    @Column(columnDefinition = "TEXT")
    private String specialRequests;

    private LocalDateTime createdAt = LocalDateTime.now();

    // Các trường mới để lưu thông tin khách hàng
    @Column(nullable = false)
    private String customerName;

    @Column(nullable = false)
    private String phone;

    @Column(nullable = false)
    private String email;

    private String nationality;

    private String checkInTimeWindow;

    @Column(columnDefinition = "TEXT")
    private String note;

    private String paymentMethod;
}
